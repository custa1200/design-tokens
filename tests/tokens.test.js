const fs = require('fs-extra');
const path = require('path');

function traverseDir(dir, arr) {
  const returnArr = arr || [];
  fs.readdirSync(dir)
    .filter(asset => asset !== '.DS_Store')
    .forEach(file => {
      const fullPath = path.join(dir, file);
      if (fs.lstatSync(fullPath).isDirectory()) {
        traverseDir(fullPath, returnArr);
      } else {
        returnArr.push(fullPath);
      }
    });

  return returnArr;
}

const jsonFiles = traverseDir(path.join(__dirname, '../tokens/'));
const jsonMerged = jsonFiles.reduce((acc, current) => {
  // eslint-disable-next-line
  const json = require(current);
  return Object.assign(acc, json);
}, {});

describe('tokens ', () => {
  test('snapshot should be matched', () => {
    expect(jsonMerged).toMatchSnapshot();
  });
});
