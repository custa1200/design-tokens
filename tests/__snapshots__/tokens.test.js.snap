// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokens  snapshot should be matched 1`] = `
Object {
  "button": Object {
    "base": Object {
      "active": Object {
        "border-width": Object {
          "value": "2",
        },
      },
      "default": Object {
        "border-width": Object {
          "value": "1",
        },
      },
    },
    "lg": Object {
      "height": Object {
        "value": "48",
      },
      "padding-x": Object {
        "value": "24",
      },
      "padding-y": Object {
        "value": "12",
      },
    },
    "md": Object {
      "height": Object {
        "value": "36",
      },
      "padding-x": Object {
        "value": "18",
      },
      "padding-y": Object {
        "value": "9",
      },
    },
    "primary": Object {
      "active": Object {
        "background-color": Object {
          "value": "{color.blue.800.value}",
        },
      },
      "default": Object {
        "background-color": Object {
          "value": "{color.blue.600.value}",
        },
        "text-color": Object {
          "value": "{color.white.value}",
        },
      },
    },
    "primary-inverse": Object {
      "active": Object {
        "background-color": Object {
          "value": "{color.grey.100.value}",
        },
        "text-color": Object {
          "value": "{color.blue.800.value}",
        },
      },
      "default": Object {
        "background-color": Object {
          "value": "{color.white.value}",
        },
        "text-color": Object {
          "value": "{color.blue.600.value}",
        },
      },
    },
    "secondary": Object {
      "active": Object {
        "border-color": Object {
          "value": "{color.blue.800.value}",
        },
        "text-color": Object {
          "value": "{color.blue.800.value}",
        },
      },
      "default": Object {
        "background-color": Object {
          "value": "{color.transparent.value}",
        },
        "border-color": Object {
          "value": "{color.blue.600.value}",
        },
        "text-color": Object {
          "value": "{color.blue.600.value}",
        },
      },
    },
    "secondary-inverse": Object {
      "default": Object {
        "background-color": Object {
          "value": "{color.transparent.value}",
        },
        "border-color": Object {
          "value": "{color.white.value}",
        },
        "text-color": Object {
          "value": "{color.white.value}",
        },
      },
    },
    "sm": Object {
      "height": Object {
        "value": "30",
      },
      "padding-x": Object {
        "value": "15",
      },
      "padding-y": Object {
        "value": "6",
      },
    },
  },
  "color": Object {
    "black": Object {
      "value": "#000000",
    },
    "blue": Object {
      "100": Object {
        "value": "#bbdefb",
      },
      "200": Object {
        "value": "#90caf9",
      },
      "300": Object {
        "value": "#64b5f6",
      },
      "400": Object {
        "value": "#42a5f5",
      },
      "500": Object {
        "value": "#2196f3",
      },
      "600": Object {
        "value": "#1e88e5",
      },
      "700": Object {
        "value": "#1976d2",
      },
      "800": Object {
        "value": "#1565c0",
      },
    },
    "grey": Object {
      "100": Object {
        "value": "#f5f5f5",
      },
      "200": Object {
        "value": "#eeeeee",
      },
      "300": Object {
        "value": "#e0e0e0",
      },
      "400": Object {
        "value": "#bdbdbd",
      },
      "500": Object {
        "value": "#9e9e9e",
      },
      "600": Object {
        "value": "#757575",
      },
      "700": Object {
        "value": "#616161",
      },
    },
    "transparent": Object {
      "value": "#00000000",
    },
    "white": Object {
      "value": "#FFFFFF",
    },
  },
  "font-family": Object {
    "primary": Object {
      "bold": Object {
        "value": "\\"Roboto-Bold\\"",
      },
      "light": Object {
        "value": "\\"Roboto-Light\\"",
      },
      "regular": Object {
        "value": "\\"Roboto-Regular\\"",
      },
    },
  },
  "font-weight": Object {
    "300": Object {
      "value": "300",
    },
    "500": Object {
      "value": "500",
    },
    "700": Object {
      "value": "700",
    },
  },
  "icon": Object {
    "lg": Object {
      "size": Object {
        "value": 36,
      },
    },
    "md": Object {
      "size": Object {
        "value": 30,
      },
    },
    "sm": Object {
      "size": Object {
        "value": 24,
      },
    },
    "xl": Object {
      "size": Object {
        "value": 40,
      },
    },
    "xs": Object {
      "size": Object {
        "value": 18,
      },
    },
  },
  "link": Object {
    "primary": Object {
      "active": Object {
        "text-color": Object {
          "metadata": Object {
            "filter": Object {
              "android": false,
              "ios": false,
            },
          },
          "value": "{color.blue.800.value}",
        },
      },
      "default": Object {
        "text-color": Object {
          "metadata": Object {
            "filter": Object {
              "android": false,
              "ios": false,
            },
          },
          "value": "{color.blue.600.value}",
        },
      },
    },
    "primary-inverse": Object {
      "default": Object {
        "text-color": Object {
          "metadata": Object {
            "filter": Object {
              "android": false,
              "ios": false,
            },
          },
          "value": "{color.white.value}",
        },
      },
    },
    "secondary": Object {
      "default": Object {
        "text-color": Object {
          "metadata": Object {
            "filter": Object {
              "android": false,
              "ios": false,
            },
          },
          "value": "{color.grey.700.value}",
        },
      },
    },
  },
  "size": Object {
    "base": Object {
      "100": Object {
        "value": "1",
      },
      "1000": Object {
        "value": "48",
      },
      "1100": Object {
        "value": "56",
      },
      "1200": Object {
        "value": "64",
      },
      "1300": Object {
        "value": "72",
      },
      "1400": Object {
        "value": "84",
      },
      "1500": Object {
        "value": "96",
      },
      "200": Object {
        "value": "2",
      },
      "300": Object {
        "value": "4",
      },
      "400": Object {
        "value": "8",
      },
      "500": Object {
        "value": "12",
      },
      "600": Object {
        "value": "16",
      },
      "700": Object {
        "value": "24",
      },
      "800": Object {
        "value": "32",
      },
      "900": Object {
        "value": "36",
      },
    },
  },
  "spacing": Object {
    "base": Object {
      "100": Object {
        "value": "1",
      },
      "200": Object {
        "value": "2",
      },
    },
  },
  "text": Object {
    "lg": Object {
      "font-size": Object {
        "value": "18",
      },
      "line-height": Object {
        "value": "1.33",
      },
    },
    "md": Object {
      "font-size": Object {
        "value": "15",
      },
      "line-height": Object {
        "value": "1.6",
      },
    },
    "sm": Object {
      "font-size": Object {
        "value": "13",
      },
      "line-height": Object {
        "value": "1.38",
      },
    },
    "xl": Object {
      "font-size": Object {
        "value": "20",
      },
      "line-height": Object {
        "value": "1.5",
      },
    },
    "xs": Object {
      "font-size": Object {
        "value": "11",
      },
      "line-height": Object {
        "value": "1.36",
      },
    },
  },
}
`;
